src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts

```diff
@@ -344,8 +458,21 @@ export class PricingDefinitionComponent implements OnInit {
         delete pricingDefinition.dimensions[dimensionKey].price;
       }
     }
+    if (this.daysOfWeekEnabled.value) {
+      pricingDefinition.restrictions.daysOfWeek = this.selectedDays.value;
+    }
+    if (!this.timeFromEnabled) {
+      delete pricingDefinition.restrictions.timeFrom;
+      delete pricingDefinition.restrictions.timeTo;
+    }
+    for (const restrictionKey in pricingDefinition.restrictions) {
+      if (!pricingDefinition.restrictions[`${restrictionKey}`]) {
```

Comment:
Remove the quote and use restrictionKey right away.

Reply:
I don't get the '!' in the if statement: if there is no restriction then you delete it!?

Reply:
we remove property when it's null or 0 because back end it;s optimized : property is not set by default

Reply:
Ok, thanks.

Reply:
yes I changed it in other places but not here, change has been done